exporter/rabbitmqexporter/integration_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package rabbitmqexporter
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/exporter/exportertest"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/testdata"
+)
+
+const (
+	username = "myuser"
+	password = "mypass"
+	vhost    = "myvhost"
+)
+
+func TestExportWithNetworkIssueRecovery(t *testing.T) {
+	testCase := []struct {
+		name  string
+		image string
+	}{
+		{
+			name:  "test rabbitmq latest",
+			image: "rabbitmq:latest",
+		},
+	}
+
+	for _, c := range testCase {
+		t.Run(c.name, func(t *testing.T) {
+			port := randPort()
+			req := testcontainers.ContainerRequest{
+				Image:        c.image,
+				ExposedPorts: []string{fmt.Sprintf("%s:5672", port)},
+				WaitingFor: wait.ForListeningPort("5672").
+					WithStartupTimeout(2 * time.Minute),
+				Env: map[string]string{
+					"RABBITMQ_DEFAULT_USER":  username,
+					"RABBITMQ_DEFAULT_PASS":  password,
+					"RABBITMQ_DEFAULT_VHOST": vhost,
+				},
+			}
+			container := getContainer(t, req)
+			defer func() {
+				err := container.Terminate(context.Background())
```

Comment:
I don't know if it matters or not (I don't know how terminate uses the context), but `context.Background()` here will be different than the rest of calls to `context.Background()` in this method. If the container isn't getting properly terminated because of this, we can save `context.Background()` in a variable and reference it in the goroutine here.

Reply:
should be okay here since I see confirmation the container terminates in the logs and there's other contrib integration tests passing a new context to `Terminate`